go/pkg/filemetadata/filemetadata_test.go

```diff
@@ -118,6 +124,87 @@ func TestComputeFilesWithXattr(t *testing.T) {
 	}
 }
 
+func TestComputeFileDigestWithXattr(t *testing.T) {
+	xattrDgName := "user.myhash"
+	overwriteXattrDgName(t, xattrDgName)
+	tests := []struct {
+		name       string
+		contents   string
+		xattrDgStr string
+		wantDgStr  string
+		wantErr    bool
+	}{
+		{
+			name:      "no-xattr",
+			contents:  "123456",
+			wantDgStr: "8d969eef6ecad3c29a3a629280e686cf0c3f5d5a86aff3ca12020c923adc6c92/6",
+		},
+		{
+			name:       "only-digest-hash",
+			contents:   "123456",
+			xattrDgStr: "1111111111111111111111111111111111111111111111111111111111111111",
+			wantDgStr:  "1111111111111111111111111111111111111111111111111111111111111111/6",
+		},
+		{
+			name:       "full-digest-(hash+size)",
+			contents:   "",
+			xattrDgStr: "1111111111111111111111111111111111111111111111111111111111111111/666",
+			wantDgStr:  "1111111111111111111111111111111111111111111111111111111111111111/666",
+		},
+		{
+			name:       "invalid-digest-hash",
+			contents:   "123456",
+			xattrDgStr: "abc",
+			wantDgStr:  "abc/6",
+			wantErr:    true,
+		},
+		{
+			name:       "invalid-full-digest",
+			contents:   "123456",
+			xattrDgStr: "666/666",
+			wantDgStr:  "666/666",
+			wantErr:    true,
+		},
+		{
+			name:       "invalid-full-digest-(extra-slash)",
+			contents:   "123456",
+			xattrDgStr: "///666",
+			wantDgStr:  "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855/0",
+			wantErr:    true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			filePath := tc.name
+			//In the Github pipeline, set xatrr for files created under /tmp folder (by running t.tempDir()) will raise an error for operation not supported
```

Comment:
Do you know why this happens?

Reply:
The symptom is when I try to set xattr for a file created under `t.TempDir()`, I will get an error message for `operation not support`. This is the same error message as what I will get if I use something not starting with `user` for the xattr key, and this only happens on the pipeline machine, I cannot locally reproduce it. 

But I see the re-client code is following the same pattern, it is using the `t.TempDir()` for log. related files(line 67), but generate the file to set xattr to only under the working directory (line 80); 

https://source.corp.google.com/foundry-x-re-client/tests/integ/xattr/xattr_digest_test.go;l=6?q=xattr&sq=&ss=piper%2FGoogle%2Ffoundry-x-re-client#:~:text=integ.SetServerAddressEnvVariable(serverAddress)-,tmpDir%20%3A%3D%20t.TempDir(),%7D,-defer%20os.RemoveAll